### PR TITLE
Fixed alpaka matmul to change alpaka api

### PR DESCRIPTION
**Pull request by erikzenker**
This PR fixes changes in the alpaka api, so that matmul compiles for the alpaka cases for the
current [alpaka develop branch](https://github.com/ComputationalRadiationPhysics/alpaka/commit/c5ab7e716ef42435b247ab26d064e2896b5fddda).

### CODE REVIEW

**Comment by erikzenker** [comment]
CI says no, and the alpaka CUDA back-ends is also not willing to cooperate :worried:

**Comment by BenjaminW3** [comment]
I am not sure but I think CI says always no because the MPI tests fail... This is known.

File: include/matmul/par/Alpaka.hpp [hunk 1]
```diff
@@ -386,6 +386,11 @@
         TElem const beta,
         TElem * const MATMUL_RESTRICT C, TSize const ldc)
     {
+        using Dim1 = alpaka::dim::DimInt<1u>;
+        using Dim2 = alpaka::dim::DimInt<2u>;
+        using Dim3 = alpaka::dim::DimInt<3u>;
```
Comment (BenjaminW3) [inline include/matmul/par/Alpaka.hpp]: Could you please remove Dim1 and Dim3 as they are not used?

**Comment by BenjaminW3** [comment]
Thanks for fixing those incompatibilities!

File: include/matmul/par/Alpaka.hpp (hunk 1 above)
Comment (erikzenker) [inline include/matmul/par/Alpaka.hpp]: fixed

**Pull request merged**